### DISCUSSION TITLE
How to deploy as a beginnner

### DISCUSSION THREAD

**Discussion by cramer-ai** [Jxck-S/plane-notify#129] [General]
Hey all - I am a newbie when it comes to coding but want to start learning. Can someone help me with the step by step of how to install / deploy this code? What software do I need and how do I install the code / run it?

Many thanks!

**Comment by Cfomodz** [comment]
Hey! Welcome to coding! Here's a step-by-step guide to get plane-notify running. This assumes you're using a Linux system (Ubuntu/Debian). If you're on Windows, I'd recommend using Docker instead (see Docker section below).

## Prerequisites

You'll need a server or computer running Linux. If you don't have one, you can use:
- A cloud server (DigitalOcean, AWS, etc.)
- A Raspberry Pi
- Windows Subsystem for Linux (WSL) on Windows
- A virtual machine

## Step 1: Install Python and Required Tools

Open your terminal and run these commands one at a time:

```bash
# Update your system
sudo apt update

# Install Python 3 and pip (Python's package manager)
sudo apt install python3
sudo apt install python3-pip

# Install Git (to download the code)
sudo apt install git

# Install Screen (lets the program run in the background)
sudo apt install screen
```

## Step 2: Install Pipenv

Pipenv manages Python dependencies for the project:

```bash
pip install pipenv
```

## Step 3: Download plane-notify

```bash
# Clone the repository
git clone -b multi --single-branch https://github.com/Jxck-S/plane-notify.git

# Move into the project directory
cd plane-notify
```

## Step 4: Install Project Dependencies

```bash
pipenv install
```

## Step 5: Install Chrome (for screenshots)

This is needed to take screenshots of planes on the map:

```bash
curl -sSL https://dl-ssl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" | sudo tee /etc/apt/sources.list.d/google.list
sudo apt update
sudo apt install google-chrome-stable
```

## Step 6: Configure the Program

### Main Configuration File

```bash
# Navigate to the configs folder
cd configs

# Copy the example files
cp mainconf.ini.example mainconf.ini
cp plane1.ini.example plane1.ini

# Edit the main config
nano mainconf.ini
```

**In `mainconf.ini`, you need to configure:**
- **SOURCE**: Set to `OPENS` for OpenSky (free) or `RpdADSBX` for ADS-B Exchange (paid)
- **RapidAPI key** (if using ADS-B Exchange RapidAPI): Get one from [RapidAPI](https://rapidapi.com/adsbx/api/adsbexchange-com1/)
- **Discord webhook** (optional): Create one in Discord Server Settings → Integrations → Webhooks
- Leave other settings as default for now

### Plane Configuration File

```bash
# Edit the plane config
nano plane1.ini
```

**In `plane1.ini`, you need to configure:**
- **ICAO**: The ICAO24 hex code of the plane you want to track (e.g., `a835af` for Elon's jet)
  - You can find ICAO codes on [ADS-B Exchange](https://globe.adsbexchange.com/) by searching for a tail number
- **TYPES**: Airport types the plane uses (leave default: `[small_airport, medium_airport, large_airport]`)
- **Discord settings**: If you want notifications:
  - Set `ENABLE = TRUE`
  - Add your webhook `URL`
  - Add a `USERNAME` (what the bot shows as)
  - Add a `TITLE` (optional text at the start of messages)

**Tip:** Press `Ctrl+X`, then `Y`, then `Enter` to save and exit nano.

## Step 7: Run the Program

```bash
# Go back to the main project directory
cd ..

# Create a screen session (so it runs in background)
screen -R plane-notify

# Start the program
pipenv run python __main__.py
```

The program should now be running! You'll see console output when it checks for the plane.

**To detach from the screen session** (let it run in background):
- Press `Ctrl+A`, then `D`

**To reattach to see the logs later:**
```bash
screen -r plane-notify
```

## Alternative: Using Docker (Easier)

If the above seems complicated, Docker is much simpler:

1. **Install Docker**: Follow instructions at [docker.com](https://docs.docker.com/get-docker/)

2. **Download plane-notify**:
   ```bash
   git clone -b multi --single-branch https://github.com/Jxck-S/plane-notify.git
   cd plane-notify
   ```

3. **Configure your files** (same as Step 6 above):
   ```bash
   cd configs
   cp mainconf.ini.example mainconf.ini
   cp plane1.ini.example plane1.ini
   nano mainconf.ini  # Edit as needed
   nano plane1.ini    # Edit as needed
   cd ..
   ```

4. **Run with Docker**:
   ```bash
   docker-compose up -d
   ```

5. **View logs**:
   ```bash
   docker logs plane-notify -f
   ```

## Finding ICAO Codes

To track a plane, you need its ICAO24 code:
1. Go to [ADS-B Exchange](https://globe.adsbexchange.com/)
2. Search for the plane's tail number (registration)
3. Click on the plane
4. Look for the "ICAO24" or "Mode S" hex code (e.g., `a835af`)

## Common Issues

- **"No module named..."**: Make sure you ran `pipenv install`
- **Chrome/ChromeDriver errors**: Make sure Chrome is installed correctly
- **No data showing**: Check that your ICAO code is correct and the plane is currently flying
- **API errors**: Verify your API keys are correct in `mainconf.ini`

## Need More Help?

- Join the Discord: https://discord.gg/groundcontrol
- Check the [README](https://github.com/Jxck-S/plane-notify/blob/multi/README.md) for more details
- Review the [example images](https://github.com/Jxck-S/plane-notify/tree/multi/ExImages) to see what output looks like

Good luck with your coding journey! 🛩️
